### PR TITLE
[#296] Split client in tcp_client and stdio_client

### LINKED ISSUE #296: Separate TCP from STDIO client
At the moment, the `els_client` module contain the logic to support both transports. The module should be splitted in two.

### PR DESCRIPTION
**Pull request by robertoaloi**
### Description

Part of a long-due cleanup on how the two transports (tcp, stdio) are used by client and server.

Fixes #296 .

### CODE REVIEW

File: src/els_client.erl [hunk 1]
```diff
@@ -3,10 +3,19 @@
...
+
 %%==============================================================================
 %% Behaviours
 %%==============================================================================
 -behaviour(gen_server).
+%% gen_server callbacks
+-export([ init/1
+        , handle_call/3
+        , handle_cast/2
+        , code_change/3
```
Comment (robertoaloi) [inline src/els_client.erl]: Removed `handle_info` since only relevant for TCP.

File: src/els_client.erl [hunk 2]
```diff
@@ -29,14 +38,7 @@
...
 
-%% gen_server callbacks
--export([ init/1
-        , handle_call/3
-        , handle_cast/2
-        , handle_info/2
-        , terminate/2
-        , code_change/3
-        ]).
+-export([ handle_responses/1 ]).
```
Comment (robertoaloi) [inline src/els_client.erl]: This is the callback which is used by the specific implementation when a message is available.

File: src/els_client.erl [hunk 3]
```diff
@@ -47,25 +49,23 @@
...
 %%==============================================================================
 %% Record Definitions
 %%==============================================================================
--record(state, { transport :: tcp | stdio
-               , connection
-               , request_id = 1
-               , pending    = []
-               , buffer   = <<>>
+-record(state, { transport_cb     :: transport_cb()
+               , transport_server :: pid()
```
Comment (robertoaloi) [inline src/els_client.erl]: This may change in the future, since the STDIO implementation is abusing this field to store the reference to the IODEVICE.

Comment (jfacorro) [reply src/els_client.erl]: I'm not sure what this comment means 😄

Comment (robertoaloi) [reply src/els_client.erl]: Yeah, not the clearest one. What I mean is that the field is named "transport_server" and it is supposed to store a Pid, but in the case of the stdio implementation we store the iodevice in there. And Dialyzer doesn't even seem to be able to figure out the discrepancy.

Comment (jfacorro) [reply src/els_client.erl]: `iodevice()`s are `pid()`s AFAIK.

**Pull request merged**